### DISCUSSION TITLE
Encrypting the data that goes into RAMJobStore

### DISCUSSION THREAD

**Discussion by askrahulsngh** [quartznet/quartznet#2474] [Q&A]
Is there in-built way of storing encrypted data into RAMJobStore? 
If not, is there a way I can override the methods on my own to store the encrypted data? Can I get pointers to the modification required for it?

**Comment by lahma** [comment]
Instead telling what you (think) you need, what about telling *why*?

Having data encrypted in memory sounds cool until you start to think more about it. Where are you storing the encryption key - in memory? So, getting the memory dump will give you both the encrypted data and the key to open it.

**Reply by askrahulsngh** [reply]
@lahma this is due to a security constraints in the type of data that we are dealing with. It includes encrypting all the data in data stores including those in memory stores. Symmetric keys will be stored in vaults. It’s expensive operation but this seems to be a constraint we are dealing with.

**Reply by lahma** [reply]
Are you storing sensitive data in JobDataMap? You could always retrieve the data inside the job from an external store that is encrypted. Quartz is in-memory when using `RAMJobStore`, it's not persisted anywhere so I don't get the need to encrypt data - it's really not a data store from my point of view, unless any List or Dictionary is considered a data store in your application.

For me this whole approach doesn't make much sense unless you are just trying to tick some box in Fancy Enterprise Application Security Checklist.

**Reply by askrahulsngh** [reply]
@lahma Thanks for a quick revert. So basically RAMJobStore is not a data store but a collection of List and Dictionary as seen [here](https://github.com/drieseng/quartznet/blob/677105b473c4e280872274667ffe22978976cf54/src/Quartz/Simpl/RAMJobStore.cs#L51C1-L60C78). This basically stores jobs and triggers as Dictionary. Adding a job is like adding a new entry into the dictionary.

**Reply by lahma** [reply]
Yes, everything stays in memory and is not persisted after process exists. Unless your jobs (written by you) write data to disk or any other persistent medium.

**Reply by askrahulsngh** [reply]
@lahma Thank you! This helps. on a different note, is there stats on RAMJobStore? By stats I'm basically referring to size needed by RAMJobStore for handling x number of jobs. Trying to create an estimate of RAM required for storing the job metadata.

**Reply by lahma** [reply]
Unfortunately no. It might be a predictable number if your jobs don't persist data after execution (in memory) but as it would be possible to job to hold megabytes of data in JobDataMap between invocations (in memory) the number would vary between use cases. You should be able to snapshot memory and create estimates based on your own usage.

**Reply by askrahulsngh** [reply]
Thank you for all the answers. Really appreciate it.

**Reply by askrahulsngh** [reply]
And thanks for this library!
